xbmc/addons/binary-addons/BinaryAddonBase.h

```diff
@@ -32,17 +29,15 @@ namespace ADDON
   public:
     explicit CBinaryAddonBase(const AddonInfoPtr& addonInfo) : m_addonInfo(addonInfo) { }
 
-    bool Create();
-
     const std::string& ID() const { return m_addonInfo->ID(); }
     const std::string& Path() const { return m_addonInfo->Path(); }
 
     TYPE MainType() const { return m_addonInfo->MainType(); }
     const std::string& MainLibName() const { return m_addonInfo->LibName(); }
 
-    bool IsType(TYPE type) const;
-    const std::vector<CBinaryAddonType>& Types() const { return m_types; }
-    const CBinaryAddonType* Type(TYPE type) const;
+    bool IsType(TYPE type) const { return m_addonInfo->IsType(type); }
```

Comment:
I like to keep lines that require knowledge of implementation details in the .cpp file. Even if it's just one line, this prevents the type from being forward-declared in the future.

Reply:
Have moved them to the cpp